### PR TITLE
Improvement/memory leak fixes

### PR DESCRIPTION
**Pull request by miquelmiro3**

### CODE REVIEW

**Review by Erick9Thor — APPROVED** [review]

File: Source/src/UI/WindowHierarchy.cpp [hunk 1]
```diff
@@ -107,6 +107,10 @@ void Hachiko::WindowHierarchy::DrawGameObject(GameObject* game_object)
         // TODO: Open options to create/destroy new object or move up down in the list of children
         if (ImGui::MenuItem("Add Game Object"))
         {
+            GameObject* selectedGameObject = App->editor->GetSelectedGameObject();
+            if (selectedGameObject == game_object)
```
Comment (armandojaga) [inline Source/src/UI/WindowHierarchy.cpp:111]: please add the braces to the if
if (condition)
{
}

**Review by armandojaga — CHANGES_REQUESTED** [review]

**Review by baransrc — APPROVED** [review]
LGTM, thanks for the good work!!

**Review by Bernatmago — APPROVED** [review]

**Review by baransrc — APPROVED** [review]
Looks good to me!! Thanks for the work!

**Review by armandojaga — APPROVED** [review]
looks good

**Review by Bernatmago — APPROVED** [review]

**Pull request merged**